### ISSUE TITLE
Can't get status code when using --error-level

### ISSUE THREAD

**Issue by memelet** [tjunnone/npm-check-updates#474]
[X]  I have read the list of known issues before filing this issue.
[X] I have searched for similiar issues before filing this issue.
* node version: v10.13.0
* npm version: v6.4.1
* npm-check-updates version: 2.15.0 and 3.0.0-alpha.3

---------------------------

No idea if this is an ncu issue or not.

When I run ncu from the command line I can get the correct error code in `$?`. But when that same command line is run via teamcity the `$?` status code is always 0. 

The script snippet looks like
```
ncu --error-level 2
check_status=$?
echo "check_status=${check_status}"
```

Any clues would be much appreciated.

**Comment by memelet** [comment]
Clearly different code is running. I've changed the command to
```
npx npm-check-updates@next --error-level INVALID
```
and it happily runs just as if the command was not specified. On the command this will result in invalid argument error.

**Comment by memelet** [comment]
Well, I was able to get a non-zero status by specifying a bogus argument flag
```
error: unknown option `--error-levelxxx'
check_status=1
```

**Comment by memelet** [comment]
I've tried every permutation I can imagine. But it is very clear that a different version is running in tc (even though the same version number is printed) or a different code path is executing (because outside of tc -- on the command line, via script, or in a subshell -- I get the final line "Dependencies not up-to-date" where in tc I never get that line).

**Comment by memelet** [comment]
By deleted you mean deleted then again `npm install`? Yes, I have done that.

But I get different results from the exact same `node_modules` -- when teamcity runs the script vs when I run it manually as the teamcity user in the same checkout directory as the build. I just can't find what would be different. I don't see anything in the code that could possibly distinguish between the two.

I have also used the 3.x alpha. Do you expect that work?

**Comment by raineorshine** [comment]
FWIW, all errors are output via this function:

```js
function programError(options, message) {
    if (options.cli) {
        print(options, message, null, 'error');
        process.exit(1);
    } else {
        throw new Error(message);
    }
}
```

https://github.com/tjunnone/npm-check-updates/blob/master/lib/npm-check-updates.js#L55-L62

**Comment by raineorshine** [comment]
v3 published: https://github.com/tjunnone/npm-check-updates/releases/tag/v3.0.0.

**Comment by Novaal** [comment]
I had the same problem, but with a gitlab runner.

The verbose log showed me the following
```
Initializing...
Running in local mode...
Finding package file data...
Waiting for package data on stdin...
Using /...../package.json
Fetching latest versions...
```

I think this is a bug in the function `getPackageDataFromFile` arround the line https://github.com/tjunnone/npm-check-updates/blob/f7cd4e2539684389a8c516f34e68bd1c53a50614/lib/npm-check-updates.js#L358 as here no Promise `pkgFile` is set, so that https://github.com/tjunnone/npm-check-updates/blob/f7cd4e2539684389a8c516f34e68bd1c53a50614/lib/npm-check-updates.js#L218 is always false.

This seems to be supported by the fact, that i cannot see none of the outputs `Dependencies not up-to-date`, `Upgraded package.json. Run 'npm install' to install new versions.` or `Run 'ncu -u' to upgrade package.json`.

My workaround is to specifiy the package.json with `--packageFile /.../package.json` so that stdin is not used.

**Comment by Novaal** [comment]
This behavior has occurred earlier and it seems not to get fixed: https://github.com/tjunnone/npm-check-updates/issues/136#issuecomment-347689346

**Comment by raineorshine** [comment]
Thank you @Novaal for the detailed explanation. It really helped me track down the issue. Hopefully it is fixed now with the v3.0.10 release!

**Comment by Novaal** [comment]
With v3.0.11 i do not need to specifying the package.json. :+1:

```
ncu --error-level 2 --loglevel verbose
Initializing...
Running in local mode...
Finding package file data...
Waiting for package data on stdin...
Checking /.../package.json
Fetching latest versions...

...

Dependencies not up-to-date
ERROR: Job failed: exit code 1
```
